packages/idyll-document/src/index.js

```diff
@@ -40,13 +42,19 @@ const getRefs = () => {
 
     const domNode = watcher.watchItem;
     const rect = domNode.getBoundingClientRect();
-    const containerNode = scrollContainer.item;
-    const containerRect = containerNode.getBoundingClientRect();
+    const percentInView = Math.max(
+      0,
+      Math.min(
+        1,
+        (watcher.bottom - scrollContainer.viewportTop) / rect.height
+      )
+    )
 
     // left and right props assume no horizontal scrolling
     refs[domNode.dataset.ref] = {
       ...bools,
       domNode,
+      percentInView: Math.round(percentInView * 100),
```

Comment:
Can we just pass the raw value for this, from 0 to 1?

Reply:
We can but the name would have to change. Open to suggestions.

Reply:
`proportionInView`, `proportionVisible`, `fractionInView`, `visibility`. interestingly there is a whole [SE question dedicated to this](https://english.stackexchange.com/questions/275734/a-word-for-a-value-between-0-and-1-inclusive), although it isn't super helpful.

Reply:
lol nice, at least we're not alone. `fractionInView` isn't bad but when I see fraction I think numerator and denominator.

Reply:
First thought was `fractionInView`.

Reply:
As a slight modification on what was already suggestion: `portionInView`. It's a rational that is always a portion. I don't see a better word than `portion` to refer to it's containment in the unit-interval, and I think `rational` is redundant because you're not in a continuous context.

Reply:
i like `portionInView`!

Reply:
Works for me. Commit incoming...